salt/modules/virt.py

```diff
@@ -1967,6 +1967,42 @@ def update(name,
                 devices_node.extend(changes[dev_type]['sorted'])
                 need_update = True
 
+    # Update the serial and the console definitions
+    if 'serial_type' in kwargs:
```

Comment:
Also add the parameters documentation to the `update` function

Reply:
According to [the libvirt domain documentation](https://libvirt.org/formatdomain.html#elementsConsole) the serial elements could be of various types and can have a source path, sometimes a mode and various other parameters.
There can also be multiple serial devices.

Along the lines of the NICs and disks, I would see a `serial` parameter as a list of dictionaries. An empty list means no serial device and the dictionary could hold properties like `type`. This way even if you don't implement all the properties of the serial devices this will be easy to extend it later on without breaking the existing behavior.

Of course the same applies for consoles.

I know these parameters are already in `virt.init()` but I think those should be deprecated in favor of the idea I described above. And thus only the news parameters would be propagated to `virt.defined()`, `virt.running()` and `virt.update()`